tests/test_actions.py

```diff
@@ -697,13 +696,37 @@ def test_implements_protocol(self) -> None:
         assert isinstance(q2, Performable)
         assert isinstance(q3, Resolvable)
 
-    def test_passthru_attribute(self):
+    @pytest.mark.parametrize("arg", (None, 1, "a"))
+    def test_quietlyperformable_raises(self, arg) -> None:
+        with pytest.raises(NotPerformable) as exc:
+            QuietlyPerformable(arg)  # type: ignore
+
+        assert str(exc.value) == ("QuietlyPerformable only works with Performable. "
+                                  "Use `Quietly` instead.")
+
+    @pytest.mark.parametrize("arg", (None, 1, "a"))
+    def test_quietlyanswerable_raises(self, arg) -> None:
+        with pytest.raises(NotAnswerable) as exc:
+            QuietlyAnswerable(arg)  # type: ignore
+
+        assert str(exc.value) == ("QuietlyAnswerable only works with Answerable. "
+                                  "Use `Quietly` instead.")
+
+    @pytest.mark.parametrize("arg", (None, 1, "a"))
+    def test_quietlyresolveable_raises(self, arg) -> None:
+        with pytest.raises(NotResolvable) as exc:
+            QuietlyResolvable(arg)  # type: ignore
+
+        assert str(exc.value) == ("QuietlyResolvable only works with Resolvable. "
+                                  "Use `Quietly` instead.")
```

Comment:
I'm not sure that we need to have 3 different examples of a not-able object, what about this instead?

```python
    @pytest.mark.parametrize(
        "SilentlyStar,AbleError", 
        (
            (SilentlyPerformable, NotPerformable),
            (SilentlyAnswerable, NotAnswerable),
            (SilentlyResolvable, NotResolvable),
    )
    def test_silentlystar_raises(self, SilentlyStar, AbleError) -> None:
        silently_name = SilentlyStar.__class__.__name__
        able_name = silently_name.replace("Silently", "")
        
        with pytest.raises(AbleError) as exc:
            SilentlyStar(None)  # type: ignore

        assert str(exc.value) == (f"{silently_name} only works with {able_name}. Use `Quietly` instead.")
```

Reply:
I think you're right about not needing the multiple parameter runs for different invalid types.   But I think it's ok to have 3 different test functions showing basically the same test explicitly.

Reply:
I am also OK with this, especially because i think the 3 explicit tests are also less code and more readable than the `parametrize` way above.